lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -2301,14 +2354,23 @@ GLStatus TFGraphLowering::lowerSequenceRegion(SequenceSESERegion *r) {
   return GLStatus::Success;
 }
 
-
 /// Given a conditional branch, produce the TF_Output for its branch condition.
 static TF_Output getCondition(CondBranchInst *condBr,
                               TFGraphLowering &lowering) {
   auto cond = condBr->getCondition();
-  auto tensorToI1 = cast<BuiltinInst>(cond);
-  assert(tensorToI1->getName().str() == "tf_tensor_to_i1" &&
-         tensorToI1->getNumOperands() == 1 &&
+  SILInstruction *tensorToI1 = nullptr;
+  // TODO: remove the case of BuiltinInst.
+  if (auto *builtinInst = dyn_cast<BuiltinInst>(cond)) {
+    assert(builtinInst->getName().str() == "tf_tensor_to_i1");
+    tensorToI1 = builtinInst;
+  } else {
+    auto *graphOpResult = cast<GraphOperationResult>(cond);
+    auto *graphOpInst = graphOpResult->getParent();
+    assert(graphOpInst->getNumResults() == 1);
+    assert(graphOpInst->getName().str() == "tf_tensor_to_i1");
+    tensorToI1 = graphOpInst;
+  }
```

Comment:
Ah, I see you'er handling tf_tensor_to_i1 here.  If you handle it above, you can remove this special case.

Reply:
It is also important to do this in the (weird) case where the tf_tensor_to_i1 is hoisted into a different block than the branch is in.

Reply:
I believe handling it here is safer and more preferable, because:
1. tf_tensor_to_i1 is conceptually more tied to how we lower cond_br, so keeping it in `getCondition()` makes the code more readable.
2. This is the same approach that we took when handling tf_tensor_to_i1 as a tfop builtin.
3. Even if it's hoisted to a different block, we will track it down via `condBr->getCondition()`.